src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -247,6 +247,10 @@ private static Assembly ResolveForKnownRequestor(string requestedAssemblyName, A
 
                     string directoryPath = Path.GetDirectoryName(requestingAssemblyFullPath);
                     string assemblyFullPath = Path.Combine(directoryPath, assemblyIdentity.Name + ".dll");
+                    if (!File.Exists(assemblyFullPath))
+                    {
+                        return null;
+                    }
 
```

Comment:
I agree with @mavasani that something may have changed. When I originally wrote this code letting exceptions escape the `AssemblyResolve` event handler didn't seem to matter, which is why the code doesn't try harder to prevent that from happening (i.e., why write code that adds complexity but doesn't do anything?).

Per discussion with @srivatsn I'm going to make a change to our event handler to prevent it from leaking *any* exceptions to the caller, but we should still take this change as well if only to avoid dealing with exceptions during debugging.

I would expect `Assembly.CodeBase` to be null for any assembly we load as we have no opportunity to specify it.